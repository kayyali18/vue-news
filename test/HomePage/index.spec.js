import { mount, shallow, createLocalVue } from '@vue/test-utils'
import VueMaterial from 'vue-material'

import HomePage from '@/pages/index.vue'
import { resolvePromises } from '../test-utils/resolvePromises'

// Create an extended 'Vue' constructor
const localVue = createLocalVue()

// Install plugins as normal
localVue.use(VueMaterial)

describe('HomePage', () => {
  // Before each test make sure we include the Vue Material lib
  let wrapper
  beforeEach(() => {
    wrapper = mount(HomePage, {
      localVue,
      mocks: {
        $axios: axios
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('to match Snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it renders newsAPI data', async () => {
    wrapper.vm.$options.asyncData({ app })
    await resolvePromises()

    expect(wrapper.html()).toContain(
      `<md-card-header><div><a>Hello</a></div></md-card-header>`
    )
  })
})
