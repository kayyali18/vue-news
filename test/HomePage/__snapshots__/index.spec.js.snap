// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage to match Snapshot 1`] = `
<main>
  <p>
    The Thinker
  </p>
   
  <button
    class="md-button md-theme-default"
    type="button"
  >
    <div
      class="md-ripple"
    >
      <div
        class="md-button-content"
      >
        Submit
      </div>
       
    </div>
  </button>
</main>
`;
